proof-systems/src/darlin/pcd/simple_marlin.rs

```diff
@@ -23,6 +23,143 @@ use std::marker::PhantomData;
 #[derive(CanonicalSerialize, CanonicalDeserialize)]
 pub struct MarlinProof<G: AffineCurve, D: Digest>(pub Proof<G::ScalarField, InnerProductArgPC<G, D>>);
 
+/// Check that vk.index_info and vk.index_comms are consistent with specified segment_size,
+/// max_domain_h_size and max_domain_k size
+pub fn is_vk_consistent_with<G: AffineCurve, D: Digest>(
+    vk: &MarlinVerifierKey<G::ScalarField, InnerProductArgPC<G, D>>,
+    segment_size: usize,
+    max_domain_h_size: usize,
+    max_domain_k_size: usize,
+) -> bool
+{
+    let num_constraints = vk.index_info.num_constraints.next_power_of_two();
+    let num_variables = vk.index_info.num_variables.next_power_of_two();
+    let domain_h_size = std::cmp::max(num_constraints, num_variables);
```

Comment:
Even if it doesn't change the logic, I'd use the next_power_of_two() to define domain_h_size. 
(For now, a domain size is always two-adic).